tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc

```diff
@@ -275,1817 +314,2459 @@ TEST(LinalgBackendEigen, add_vector)
+	               -0.1, 0.0, 0.1};
 
-	SGMatrix<float64_t> m(data, n, n, false);
+	SGMatrix<ST> m(data, n, n, false);
 
 	center_matrix(m);
 
 	for (index_t i = 0; i < (index_t)m.size(); ++i)
-		EXPECT_NEAR(m[i], result[i], 1e-8);
+		EXPECT_NEAR(m[i], result[i], 1e-7);
 }
 
-TEST(LinalgBackendEigen, SGMatrix_cholesky_llt_lower)
+template <typename ST>
+void check_SGMatrix_cholesky_llt_lower()
 {
 	const index_t size=2;
-	SGMatrix<float64_t> m(size, size);
+	SGMatrix<ST> m(size, size);
+	typedef Matrix<ST, Dynamic, Dynamic> Mxx; // need to adapt the Eigen::Matrix to ST type
 
 	m(0,0)=2.0;
 	m(0,1)=1.0;
 	m(1,0)=1.0;
 	m(1,1)=2.5;
 
 	//lower triangular cholesky decomposition
-	SGMatrix<float64_t> L = cholesky_factor(m);
+	SGMatrix<ST> L = cholesky_factor(m);
+
 
-	Map<MatrixXd> map_A(m.matrix,m.num_rows,m.num_cols);
-	Map<MatrixXd> map_L(L.matrix,L.num_rows,L.num_cols);
+	Map<Mxx> map_A(m.matrix,m.num_rows,m.num_cols);
+	Map<Mxx> map_L(L.matrix,L.num_rows,L.num_cols);
 	EXPECT_NEAR((map_A-map_L*map_L.transpose()).norm(),
-		0.0, 1E-15);
+		0.0, 1E-6); // need to change the treshold to work with floats
 	EXPECT_EQ(m.num_rows, L.num_rows);
 	EXPECT_EQ(m.num_cols, L.num_cols);
 }
 
-TEST(LinalgBackendEigen, SGMatrix_cholesky_llt_upper)
+template <typename ST>
+void check_SGMatrix_cholesky_llt_upper()
 {
-	const index_t size=2;
-	SGMatrix<float64_t> m(size, size);
+    typedef Matrix<ST, Dynamic, Dynamic> Mxx;
 
-	m(0,0)=2.0;
-	m(0,1)=1.0;
-	m(1,0)=1.0;
-	m(1,1)=2.5;
+    const index_t size=2;
+    SGMatrix<ST> m(size, size);
+
+    m(0,0)=2.0;
+    m(0,1)=1.0;
+    m(1,0)=1.0;
+    m(1,1)=2.5;
+
+    //upper triangular cholesky decomposition
+    SGMatrix<ST> U = cholesky_factor(m,false);
+
+    Map<Mxx> map_A(m.matrix,m.num_rows,m.num_cols);
+    Map<Mxx> map_U(U.matrix,U.num_rows,U.num_cols);
+    EXPECT_NEAR((map_A-map_U.transpose()*map_U).norm(),
+    0.0, 1E-6); // need to change the treshold to work with floats
+    EXPECT_EQ(m.num_rows, U.num_rows);
+    EXPECT_EQ(m.num_cols, U.num_cols);
+}
+
+template <typename ST>
+void check_SGMatrix_cholesky_rank_update_upper()
+{
+    typedef Matrix<ST, Dynamic, Dynamic> Mxx;
+    typedef Matrix<ST, Dynamic, 1> Vx;
+
+    const index_t size = 2;
+    ST alpha = 1;
+    SGMatrix<ST> A(size, size);
+    SGMatrix<ST> U(size, size);
+    SGVector<ST> b(size);
+    Map<Mxx> A_eig(A.matrix, size, size);
+    Map<Mxx> U_eig(U.matrix, size, size);
+    Map<Vx> b_eig(b.vector, size);
+
+    U(0, 0) = 2.0;
+    U(0, 1) = 1.0;
+    U(1, 1) = 2.5;
+    b[0] = 2;
+    b[1] = 3;
+    A_eig = U_eig.transpose() * U_eig;
+
+    auto A2 = A.clone();
+    Map<Mxx> A2_eig(A2.matrix, A2.num_rows, A2.num_cols);
+    A2(0, 0) += b[0] * b[0];
+    A2(0, 1) += b[0] * b[1];
+    A2(1, 0) += b[1] * b[0];
+    A2(1, 1) += b[1] * b[1];
+
+    cholesky_rank_update(U, b, alpha, false);
+    EXPECT_NEAR((A2_eig - U_eig.transpose() * U_eig).norm(), 0.0, 1e-5);  // need to change the treshold to work with floats
```

Comment:
minor style: we do comments in a line above usually (+ a whitespace line above that)